src/main/java/com/google/sps/data/Keywords.java

```diff
@@ -36,27 +40,31 @@
 
   // We don't want any terms that are far too un-salient to include in the keywords list.
   // 0.2 is a good minimum threshold as, especially when extracting saliency from a list of
-  // label, many of the high-relevancy terms end up only getting assigned a salience of 0.2.
-  private static final double MIN_SALIENCE_THRESHOLD = 0.1;
+  // label, many of the high-relevancy terms end up only getting assigned a salience of 0.08.
+  private static final double MIN_SALIENCE_THRESHOLD = 0.08;
 
-  // For labels, we want high-confidence labels. Many of the labels below 0.70 seem to end
+  // For labels, we want high-confidence labels. Many of the labels below 0.60 seem to end
   // up being only tangetially related to the contents of the picture.
   private static final double MIN_SCORE_THRESHOLD = 0.60;
 
   /**
-   * @return a list of the 5 most salient keywords
+   * @return a list of the 10 most salient keywords from the datastore in accordance to the keys
    */
-  public static List<String> getKeywords(String keyString) {
-    try {
-      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-      Entity result = datastore.get(KeyFactory.stringToKey(keyString));
-    
-      Collection<String> keywordCollection = (Collection<String>) result.getProperty("keywords");
-      List<String> keywordList = keywordCollection.stream().limit(MAX_NUM_KEYWORDS).collect(Collectors.toList());
-      return keywordList;
-    } catch (EntityNotFoundException ex) {
-      return new ArrayList<>();
-    }
+  public static Collection<String> getKeywords(String keyString) {
+    UserService userService = UserServiceFactory.getUserService();
+    Map<String, Collection<String>> allKeywords = getKeyKeywordMap(userService.getCurrentUser().getUserId());
+        
+    Collection<String> keywordCollection = allKeywords.get(keyString);
+    List<String> keywordList = keywordCollection.stream().limit(MAX_NUM_KEYWORDS).collect(Collectors.toList());
```

Comment:
[Optional] You cal add this limit to the datastore query itself.

Reply:
I guess I could move this into the addToDatastore method itself, so I can just grab and return.